server/src/GraphQL/Mutations/Checkout.php

```diff
@@ -26,15 +28,30 @@ public function checkout($rootValue, array $args, GraphQLContext $context, Resol
         $cartService = App::make(CartService::class);
         $sessionService = App::make(SessionService::class);
 
-        if( !$sessionService->exists() ){
+        if (!$sessionService->exists()) {
             return false;
         }
 
-        if( $cartService->hasItems() ){
+        if ($cartService->hasItems()) {
+
+            // Fire the before checkout event
+            $eventService = EventService::triggerEvent(
```

Comment:
@AlexeiaW so at time of checkout we'll need to be passing the payment provider, probably as a string, like "stripe_standard" or whatever this particular flavour of Stripe may be called. This would pass into the event service. Then in Stripe's action, you can check if this provider string matches, or just return an empty action. 

So in this case, if we had a chain of 10 providers, this would demonstrate the choice made by the user on the front end. They would have picked from a bullet list of payment options, and picked Stripe in this case.

Reply:
@luke-siedle I implemented this